### ISSUE TITLE
"more" block keeps attribute and empty string when text is cleared and saved

### ISSUE THREAD

**Issue by amdrew** [WordPress/gutenberg#5583]
## Issue Overview

On the latest master branch and when using the "more" block, clearing the text keeps the `customText` attribute in the `post_content` as an empty string:

```
<!-- wp:more {"customText":""} -->
<!--more-->
<!-- /wp:more -->
```

It also doesn't "reset" the tag back to its original state of "Read more" (as defined as the default text in the block's code):

<img width="713" alt="screen shot 2018-03-13 at 10 02 12 pm" src="https://user-images.githubusercontent.com/52581/37333265-cc156a68-270d-11e8-82c6-83dc3860f88f.png">

## Steps to Reproduce

1. Add a "more" block and save. Image:

    <img width="701" alt="screen shot 2018-03-13 at 9 58 45 pm" src="https://user-images.githubusercontent.com/52581/37333540-747c282c-270e-11e8-957a-a88b988cc2f5.png">

    The `post_content` is correctly saved as:

    ```
    <!-- wp:more -->
    <!--more-->
    <!-- /wp:more -->
    ```

2. Now, change the text to `Read more!` and save. The `post_content` now has a `customText` attribute as expected:

    ```
    <!-- wp:more {"customText":"Read more!"} -->
    <!--more Read more!-->
    <!-- /wp:more -->
    ```

3. Clear the text and save. The block shows no text, and the `customText` attribute is saved as an empty string in the `post_content`:

    <img width="713" alt="screen shot 2018-03-13 at 10 02 12 pm" src="https://user-images.githubusercontent.com/52581/37333940-74638398-270f-11e8-99b6-e77bc69e3233.png">

    ```
    <!-- wp:more {"customText":""} -->
    <!--more-->
    <!-- /wp:more -->
    ```

## Expected Behavior

1. When clearing the text, I'd expect the text to default back to "Read more". This is the text defined in the code as the default text. It's also the text you see when you first add the block.

2. When the text is cleared and the post/page updated, the attribute should be removed from the `post_content`. It should match the `post_content` when you first insert the "more" block and save. I.e.

```
<!-- wp:more -->
<!--more-->
<!-- /wp:more -->
```

**Comment by amdrew** [comment]
PR: https://github.com/WordPress/gutenberg/pull/5584

**Comment by aduth** [comment]
>It also doesn't "reset" the tag back to its original state of "Read more" (as defined as the default text in the block's code):

An alternative interpretation here is that there is a bug, but it's that we don't accurately reflect the empty string in the serialization.

In the classic editor, if one enters this into the Text tab:

```html
<!--more -->
```

...and views the post on the blog listing, the More tag exists, but is obviously empty and not clickable.

Whether this would ever serve any real purpose (vs. being more confusing) is another question.

**Comment by aduth** [comment]
Closed by #5584

For future reference, you can use keywords like "FIxes #5583" to auto-close related issues from pull request merge. See: https://help.github.com/articles/closing-issues-using-keywords/
